### DISCUSSION TITLE
[GitHub Actions] - Jobs crashing my apps

### DISCUSSION THREAD

**Discussion by mhenrk** [community/community#36610] [Actions]
Good evening folks,
I'm implementing two systems on amazon, using the following configuration:

AWS EC 2 Configuration
Ubuntu Linux Server 22.04LTS - 30Gb SSD - 1 GB memory (Free Tier)

On that server will be a node v16 application (Api), a MySql database (running in docker) and a reactjs application (using vitejs) and PM2 to persist

It happens that I need the actions for the rest of the team to be able to make updates to the project, but when configuring my self-hosted runner (creating the nodejs action and then performing it changing to self-hosted, after the configurations) the dependency installation and build job (when even the build) crashes to the point of losing my communication with the server on aws and needs to interrupt the instance to use it again.

Is there any possibility to solve this problem? It can be some kind of configuration? Memory ?
could someone guide me on a good architecture (that can be made for free by amazon) for this purpose?
thanks for all.

**Comment by vitalyk-multinarity** [comment]
I suggest reviewing /var/log/syslog - you'll see OOM killer events if they happened.
1GB RAM maybe not be enough for all your stuff.

**Comment by ossanna16** [comment]
Hey there @mhenrk and thank you for asking a good question! I'm glad you found our community 🙂

**Comment by dharunkpr** [comment]
Hello @mhenrk ,

You can write the self hosted command into a shell scripting file on the server and give it execution access, and execute the script file using  "appleboy/ssh-action@master" uses.
 
And the self-hosted pipeline consumes too much storage for artifacts.
So you can deploy through the scripting file with github Hosted Service.
